astropy/modeling/core.py

```diff
@@ -2411,6 +2477,8 @@ def _check_inputs_and_outputs(mcls, operator, left, right):
         if operator == '|':
             inputs = left.inputs
             outputs = right.outputs
+            inputs_map = {inp: (left, inp) for inp in left.inputs}
+            outputs_map = {out: (right, out) for out in right.outputs}
```

Comment:
So do these update `_inputs_map` and `_outputs_map`?

Reply:
Actually, looking at the objects id I think they don't. The reason is that the compound model makes a copy of the submodels. I can't say immediately if this may become a problem.

Reply:
No they don't update, and this casues major issues with the fitting. However, I can see no way to construct these mappings on-the-fly. I think we can postpone this to the fitting PR, as it's all very deep internals anyway?

Reply:
Yeah, the fitting can be done separately. But this causes the model to become out of sync with its submodels, for example:
```
g1 = Gaussian1D()
offx = Shift()
m = g1 | offx
g1.input_units_strict=True
```
`m[0].input_units_strict`  is now different from `m.input_units_strict` which I'm guessing was not the intention.

Not sure how important this is.